OpenRA.Game/Settings.cs

```diff
@@ -300,6 +302,52 @@ public void Save()
 			root.WriteToFile(settingsFile);
 		}
 
+		public static string SanitizedName(string dirty)
+		{
+			// reserved characters for MiniYAML and JSON
+			var disallowedChars = new char[] { '#', '@', ':', '\n', '\t', '[', ']', '{', '}', '"', '`' };
+
+			var clean = dirty;
+			if (string.IsNullOrEmpty(dirty))
+				clean = "";
```

Comment:
This will probably need to have some non-empty default value (`Newbie` is the default player name, so probably that), otherwise the sanitation in this case doesn't really work.